app/furniture/marketplace/management_component.rb

```diff
@@ -16,9 +16,12 @@ def button(location, icon:)
       label, href = if location.is_a?(Symbol)
         [t("marketplace.marketplace.#{location}.link_to"),
           marketplace.location(location)]
-      else
+      elsif location[:child].to_s.pluralize == location[:child].to_s
```

Comment:
Non-blocking:
I feel that this check could be later extracted to some sort of `is_plural?` helper function, but I don't know if that would somewhat defeat the purpose behind the plural/singular design "constraints".

Reply:
yea, I think this is quite clugey; and a [`Refinement`](https://ruby-doc.org/core-3.1.0/Refinement.html) that allows strings and symbols to tell us whether they are singular or plural would be much preferred; especially if we see this kind of check emerging as a pattern.